docs/events_handling.md

```diff
@@ -0,0 +1,89 @@
+
+Official GitHub documentation: [PullRequest event](https://developer.github.com/v3/activity/events/types/#pullrequestevent)
+
+* pull_request.opened - the first event which occurs when a new pull request is created. It's a self-explanatory
+
+* pull_request.reopened - notifies Precaution to start a new scan on a reopened pull request
+
+* pull_request.synchronize - occurs when you add a commit to already opened pull request.
+We discussed it and added it in pull request number [#113](https://github.com/vmware/precaution/pull/113)
+
+### Check run/suite events
+
+Official GitHub documentation:
+- [CheckRun event](https://developer.github.com/v3/activity/events/types/#checkrunevent)
+- [CheckSuite event](https://developer.github.com/v3/activity/events/types/#checksuiteevent)
+
+You had already opened a pull request and you are at the pull request page.
+Then you navigate to the "Checks" tab.
+If the Precaution check had failed you will see "Re-run all checks", "Re-run" and
+"Re-run failed checks" buttons.
+
+* check_run.rerequested - occurs when you click the "Re-run" button.
+This will tell Precaution to start a new check run.
+
+* check_suite.rerequested - happens when you click "Re-run all checks" 
+or "Re-run failed checks" buttons. 
+This notifies Precaution and the other GitHub apps 
+that they have to start a new check run.
+
+## Events workflows
+
+This is a small paragraph about the events workflow we are receiving with and posting to GitHub.
+It doesn't aim to explain in details how Precaution works.
+If you want to read more details how Precaution works you can look at our [Architecture]().
+
+There are numerous use cases but the triggered events and their order are alike.
+In the text below "F" equals "forked repository" and "U" means upstream repository.
+
+### Pull request from F to U or from U to itself
+
+1. `pull_request.opened` webhook received
+2. `POST /repos/:owner/:repo/check-runs` - API call to create check_run
+3. `GET /repos/:owner/:repo/contents/:.precaution.yaml` - API call to get `.precaution.yaml` config file
+4. `GET /repos/:owner/:repo/pulls/:number/files` - API call to get list of modified files
+5. `GET /repos/:owner/:repo/contents/:path` - multiple API calls to get the modified files
+6. `PATCH /repos/:owner/:repo/check-runs/:check_run_id` - API call which sends the results
+
+### Add commit to already opened pull request
+
+1. `pull_request.synchronize webhook` webhook received
+2. `POST /repos/:owner/:repo/check-runs` - API call to create check_run
+3. `GET /repos/:owner/:repo/contents/:.precaution.yaml` - API call to get `.precaution.yaml` config file
+4. `GET /repos/:owner/:repo/pulls/:number/files` - API call to get list of modified files
+5. `GET /repos/:owner/:repo/contents/:path` - multiple API calls to get the modified files
+6. `PATCH /repos/:owner/:repo/check-runs/:check_run_id` - API call which sends the results
+
+If you have already opened a pull request and you navigate to the "Checks" tab then
```

Comment:
Run-on sentence

Reply:
This duplicates information from the Check*Event section, no need to repeat it here.

Reply:
Yea, I will remove it. Probably we don’t need it.